.github/workflows/main.yml

```diff
@@ -188,6 +188,12 @@ jobs:
         role-to-assume: ${{ secrets.AWS_UPLOAD_ROLE_ARN }}
     - name: Set PreRelease Version
       run: echo "GORELEASER_CURRENT_TAG=v$(pulumictl get version --language generic)" >> $GITHUB_ENV
+    - name: Check PreRelease Version and Exit if not Alpha, Beta, or RC
+      run: |
+        if [[ ! "${GORELEASER_CURRENT_TAG}" =~ -(alpha|beta|rc) ]]; then
+          echo "Error: GORELEASER_CURRENT_TAG does not contain -alpha, -beta, or -rc"
+          exit 1
+        fi
```

Comment:
I'm assuming this is consistent with other workflows?

Reply:
On the ecosystem & providers team, we limit pushes so that only pull requests can push to main, which prevents us from pushing already tagged commits. (Probably, as I write this I realize it's possible but not obvious how to do that.)

Reply:
Looking at audit logs, it looks like this is just defensive coding, happy to remove.

Reply:
It looks like the PR was merged, `main.yml` workflow began running, and the release was tagged shortly after; but before the publish *job* ran. When the publish job ran, it picked up that the commit was already tagged 0.7.2.